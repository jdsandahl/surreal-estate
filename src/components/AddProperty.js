import React, { useState } from "react";
import "../styles/AddProperty.css";

import Alert from "../components/Alert";
import PropertyForm from "../components/PropertyForm";
import postProperty from "../services/postProperty";

const AddProperty = () => {
  const initialState = {
    fields: {
      title: "",
      city: "Manchester",
      type: "Flat",
      bedrooms: "1",
      bathrooms: "1",
      price: 100000,
      email: "yourName@example.com",
    },
  };

  const [fields, setFields] = useState(initialState.fields);

  const handleAddProperty = (event) => {
    event.preventDefault();
    postProperty(fields);
  };

  const handleFieldChange = (event) => {
    const changedField = event.target.name;
    const newValue = event.target.value;
    setFields({ ...fields, [changedField]: newValue });
  };

  return (
    <div className="AddProperty" data-testid="AddProperty">
      <Alert message="dummy test message" />
      <PropertyForm
        handleAddProperty={handleAddProperty}
        handleFieldChange={handleFieldChange}
        title={fields.title}
        city={fields.city}
        type={fields.type}
        bedrooms={fields.bedrooms}
        bathrooms={fields.bathrooms}
        price={fields.price}
        email={fields.email}
      />        
    </div>
  );
};

export default AddProperty;
