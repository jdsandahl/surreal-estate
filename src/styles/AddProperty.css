.AddProperty {
  padding: 20px;
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: var(--secondary-light-color);
  height: 100vh;
}

.add-property-form {
  display: flex;
  flex-direction: column;
  border: 1px solid var(--tertiary-dark-color);
  box-shadow: 0 4px 8px 0 var(--tertiary-dark-color);
  padding: 25px;
  min-width: 300px;
  background-color: var(--primary-light-color);
}

.form-label {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  padding: 5px;
  font-weight: bold;
}

.form-input {
  flex: 0 1 auto;
  align-items: center;
  margin: 5px;
  padding: 5px;
  border: 1px solid var(--tertiary-dark-color);
  border-radius: 5px;
}


.form-select {
  flex: 0 1 auto;
  margin: 5px;

  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: flex-start;
  padding-right: 5px;

  appearance: none;
  background-image: var(--primary-dark-color);
  border: 1px solid var(--tertiary-dark-color);
  border-radius: 5px;
}

.form-options {
  flex: 0 1 auto;
  margin: 5px;
  border: none;
}

.add-property-button {
  margin: 5px;
  padding: 5px;
  background-color: var(--primary-dark-color);
  color: var(--light-text);
  font-weight: 900;
  font-size: medium;
}

.add-property-button:hover {
  background-color: var(--secondary-dark-color);
  color: var(--light-text);
}