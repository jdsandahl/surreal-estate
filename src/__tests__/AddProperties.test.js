import React from "react";
import { render } from "@testing-library/react";
import AddProperty from "../components/AddProperty";

test('renders AddProperty component', () => {
    const { getByText } = render(<AddProperty />);

    const pageElement = getByText(/add property page/i);

    expect(pageElement).toBeInTheDocument();
});