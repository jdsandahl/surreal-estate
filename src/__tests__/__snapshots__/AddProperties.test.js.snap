// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddProperty the page renders correctly against snapshot 1`] = `
<DocumentFragment>
  <div
    class="AddProperty"
    data-testid="AddProperty"
  >
    <form
      class="add-property-form"
      data-testid="property-form"
    >
      <label
        class="form-label"
      >
        Title:
        <input
          class="form-input"
          id="title"
          name="title"
          value=""
        />
      </label>
      <label
        class="form-label"
      >
        City:
        <div
          class="form-select"
        >
          <select
            class="form-options"
            id="city"
            name="city"
          >
            <option
              value="Manchester"
            >
              Manchester
            </option>
            <option
              value="Leeds"
            >
              Leeds
            </option>
            <option
              value="Sheffield"
            >
              Sheffield
            </option>
            <option
              value="Liverpool"
            >
              Liverpool
            </option>
          </select>
        </div>
      </label>
      <label
        class="form-label"
      >
        Type:
        <div
          class="form-select"
        >
          <select
            class="form-options"
            id="type"
            name="type"
          >
            <option
              value="Flat"
            >
              Flat
            </option>
            <option
              value="Detached"
            >
              Detached
            </option>
            <option
              value="Semi-Detached"
            >
              Semi-Detached
            </option>
            <option
              value="Terraced"
            >
              Terraced
            </option>
            <option
              value="End of Terrace"
            >
              End of Terrace
            </option>
            <option
              value="Cottage"
            >
              Cottage
            </option>
            <option
              value="Bungalow"
            >
              Bungalow
            </option>
          </select>
        </div>
      </label>
      <label
        class="form-label"
      >
        Bedrooms:
        <div
          class="form-select"
        >
          <select
            class="form-options"
            id="bedrooms"
            name="bedrooms"
          >
            <option
              value="1"
            >
              1
            </option>
            <option
              value="2"
            >
              2
            </option>
            <option
              value="3"
            >
              3
            </option>
            <option
              value="4"
            >
              4
            </option>
            <option
              value="5"
            >
              5+
            </option>
          </select>
        </div>
      </label>
      <label
        class="form-label"
      >
        Bathrooms:
        <div
          class="form-select"
        >
          <select
            class="form-options"
            id="bathrooms"
            name="bathrooms"
          >
            <option
              value="1"
            >
              1
            </option>
            <option
              value="2"
            >
              2
            </option>
            <option
              value="3"
            >
              3
            </option>
            <option
              value="4"
            >
              4
            </option>
            <option
              value="5"
            >
              5+
            </option>
          </select>
        </div>
      </label>
      <label
        class="form-label"
      >
        Price (Â£):
        <input
          class="form-input"
          id="price"
          min="0"
          name="price"
          step="1"
          type="number"
          value="100000"
        />
      </label>
      <label
        class="form-label"
      >
        Email:
        <input
          class="form-input"
          id="email"
          name="email"
          type="email"
          value="yourName@example.com"
        />
      </label>
      <button
        class="add-property-button"
        type="submit"
      >
        Add
      </button>
    </form>
  </div>
</DocumentFragment>
`;
